src/java.desktop/share/classes/javax/print/attribute/standard/OutputBin.java

```diff
@@ -0,0 +1,201 @@
+ * (<a href="https://ftp.pwg.org/pub/pwg/candidates/cs-ippoutputbin10-20010207-5100.2.pdf">
+ * PDF</a>) of IPP 1.1. The category name returned by {@code getName()} is the
+ * IPP attribute name. The enumeration's integer value is the IPP enum value.
+ * The {@code toString()} method returns the IPP string representation of the
+ * attribute value.
+ */
+public sealed class OutputBin extends EnumSyntax implements PrintRequestAttribute, PrintJobAttribute permits CustomOutputBin {
+
+    @Serial
+    private static final long serialVersionUID = -3718893309873137109L;
+
+    /**
+     * The top output bin in the printer.
+     */
+    public static final OutputBin TOP = new OutputBin(0);
+
+    /**
+     * The middle output bin in the printer.
+     */
+    public static final OutputBin MIDDLE = new OutputBin(1);
+
+    /**
+     * The bottom output bin in the printer.
+     */
+    public static final OutputBin BOTTOM = new OutputBin(2);
+
+    /**
+     * The side output bin in the printer.
+     */
+    public static final OutputBin SIDE = new OutputBin(3);
+
+    /**
+     * The left output bin in the printer.
+     */
+    public static final OutputBin LEFT = new OutputBin(4);
+
+    /**
+     * The right output bin in the printer.
+     */
+    public static final OutputBin RIGHT = new OutputBin(5);
+
+    /**
+     * The center output bin in the printer.
+     */
+    public static final OutputBin CENTER = new OutputBin(6);
+
+    /**
+     * The rear output bin in the printer.
+     */
+    public static final OutputBin REAR = new OutputBin(7);
+
+    /**
+     * The face up output bin in the printer.
+     */
+    public static final OutputBin FACE_UP = new OutputBin(8);
+
+    /**
+     * The face down output bin in the printer.
+     */
+    public static final OutputBin FACE_DOWN = new OutputBin(9);
+
+    /**
+     * The large-capacity output bin in the printer.
+     */
+    public static final OutputBin LARGE_CAPACITY = new OutputBin(10);
+
+    /**
+     * Construct a new output bin enumeration value with the given integer
+     * value.
+     *
+     * @param value Integer value
+     */
+    protected OutputBin(int value) {
+        super(value);
+    }
+
+    /**
+     * The string table for class {@code OutputBin}.
+     */
+    private static final String[] myStringTable = {
+            "top",
+            "middle",
+            "bottom",
+            "side",
+            "left",
+            "right",
+            "center",
+            "rear",
+            "face-up",
+            "face-down",
+            "large-capacity",
+    };
+
+    /**
+     * The enumeration value table for class {@code OutputBin}.
+     */
+    private static final OutputBin[] myEnumValueTable = {
+            TOP,
+            MIDDLE,
+            BOTTOM,
+            SIDE,
+            LEFT,
+            RIGHT,
+            CENTER,
+            REAR,
+            FACE_UP,
+            FACE_DOWN,
+            LARGE_CAPACITY,
```

Comment:
I updated the `PrintService.isAttributeValueSupported()` method that it always returns true for CustomOutputBin class and returns true for an OutputBin constant only if its original name (ipp case) or the name without `-` sign (cups case) equals to one of the supported output bin choice name.

Reply:
It is easy to add new constants to a public API but there is no way to remove them.
OutpuBin constants work only if they are listed among the supported output bins.
There is always a way to lookup `tray-n` output bin through the supported ones and it would work in the same way if it was included as a predefined constant.

Reply:
If it's only about adding tray-1 to tray-11 in the above list and not about whole lot of changes in other areas, so that supported list contains those "tray-N" values, I guess we should add them..
I guess isSupportedAttributedValues(tray-N) will not return false in that case..

Reply:
Suppose `getSupportedAttributeValues(OutputBin.class, null, null)` returns [`top`, `face-down`] output bins.
`isSupportedAttributedValues(...)` returns `true` for `OutputBin.TOP` and `OutputBin.FACE_DOWN` constants and `false` for `OutputBin.TRAY_1` if it was added in current implementation.

To make a one to one correspondence between `OutputBin.TOP` and `OutputBin.TRAY_1` constants it needs to add some logic and may be even make them equal in terms of `OutputBin` class which extends `EnumSyntax` class. 
It is not clear should such logic be provided in the current release or it is better to provide its proper implementation in the next release if necessary.

Reply:
ok